packages/phoenix-evals/src/phoenix/evals/evaluators.py

```diff
@@ -717,8 +805,30 @@ def word_count(text: str) -> int:
         The decorator automatically handles conversion to a valid Score object.
     """
 
+    # Resolve kind/source and deprecations once at decoration time
+    if kind is not None and source is not None and kind != source:
+        raise ValueError("Provide only one of 'kind' or 'source' (they differ). Use 'kind'.")
+    # Resolve final kind with backward compatibility for deprecated `source`
+    # If neither is provided, default to "code".
+    resolved_kind: KindType = (
+        kind if kind is not None else (source if source is not None else "code")
+    )
+    if kind is None and source is not None:
+        warnings.warn(
+            "create_evaluator 'source' is deprecated; use 'kind' instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if resolved_kind == "heuristic":
+        warnings.warn(
+            "kind='heuristic' is deprecated; use kind='code' instead.",
```

Comment:
Should this warning mention that it is going to change the kind to "code" automatically as well?